src/views/Home.jsx

```diff
@@ -1,13 +1,59 @@
 import './Home.css';
 import { SingleList } from '../components';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { createList } from '../api';
+
+export function Home({ data, setListPath, userId, userEmail }) {
+	const [listName, setListName] = useState('');
+	const navigate = useNavigate();
+
+	const handleChange = (event) => {
+		setListName(event.target.value);
+	};
+
+	const handleSubmit = async (event) => {
+		event.preventDefault();
+		const currentLists = data.map((list) => {
+			return list.name.toLowerCase();
+		});
+
+		if (currentLists.includes(listName.toLowerCase())) {
+			alert('The list already exists. Please enter a different name.');
+			setListName('');
+			return;
+		}
+
+		try {
+			const response = await createList(userId, userEmail, listName);
+			// console.log('check response', response);
+			setListPath(response);
```

Comment:
Nit: if the `listPath` is returned from `createList`, I would name the variable `listPath` to make it more clear
```suggestion
			const listPath = await createList(userId, userEmail, listName);
			// console.log('check response', response);
			setListPath(listPath);
```